### DISCUSSION TITLE
Can't build `osu.Game.Tests.{Android,iOS}`

### DISCUSSION THREAD

**Discussion by Susko3** [ppy/osu#16451] [Q&A] [answered]
Failure at:
https://github.com/ppy/osu/blob/70f56cd0c9f53c7fc13443318dac57ae46f7468b/osu.Game.Tests/Visual/Ranking/TestSceneResultsScreen.cs#L206

![image](https://user-images.githubusercontent.com/16479013/149543647-bb74db90-3e51-4042-b8c4-81e7efdc9b47.png)

Looks like the type-less class is not available for Xamarin: https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskcompletionsource?view=net-6.0#applies-to

Not sure what's the correct type to use here, but this seems to work (test passes):

```diff
diff --git a/osu.Game.Tests/Visual/Ranking/TestSceneResultsScreen.cs b/osu.Game.Tests/Visual/Ranking/TestSceneResultsScreen.cs
index 666cbf02b5..4fc5cfe37e 100644
--- a/osu.Game.Tests/Visual/Ranking/TestSceneResultsScreen.cs
+++ b/osu.Game.Tests/Visual/Ranking/TestSceneResultsScreen.cs
@@ -203,7 +203,7 @@ public void TestFetchScoresAfterShowingStatistics()
         {
             DelayedFetchResultsScreen screen = null;

-            var tcs = new TaskCompletionSource();
+            var tcs = new TaskCompletionSource<object>();

@@ -218,7 +218,7 @@ public void TestFetchScoresAfterShowingStatistics()

             AddAssert("no fetch yet", () => !screen.FetchCompleted);

-            AddStep("allow fetch", () => tcs.SetResult());
+            AddStep("allow fetch", () => tcs.SetResult(this));

             AddUntilStep("wait for fetch", () => screen.FetchCompleted);
```

**Comment by frenzibyte** [comment]
`TaskCompletionSource<object>` seems fine (`SetResult` can just have null passed to it?), should also include an inline comment explaining that so someone doesn't silently break it in the future or otherwise.

Note that even with that change, `osu.Game.Tests` is broken beyond repair for iOS due to use of Moq which cannot be AOT'd (see https://github.com/moq/moq4/issues/102).

**Reply by peppy** [reply]
I'd probably use `<bool>` as we do that in a few other places.

**Reply by Susko3** [reply]
So pass `true` for `SetResult`?

**Reply by Susko3** [reply]
Should I add it to banned symbols?
```
T:System.Threading.Tasks.TaskCompletionSource;Don't use TaskCompletionSource(). Use TaskCompletionSource<TResult>() as the former is not available on Xamarin.
```

**Reply by peppy** [reply]
Yep, `true`. Not sure about adding to `BannedSymbols`. Probably not important enough to do so.

**Comment by Susko3** [comment, marked as answer]
https://github.com/ppy/osu/pull/16452
